### ISSUE TITLE
Partner request issues on test site

### ISSUE THREAD

**Issue by nbaertlein** [USGS-WiM/whispersservices_django#380]
Issues on user account requests that on the surface appear related to USGS-WiM/whispers#371 but I think is more.  
Partner user account requests are not writing to table whispers_userchangerequest and therefore not to whispers_notifications requesting a role change.   Completing the form also throws a toast message stating the comment cannot be blank, but I can click 'Submit' again and it goes through.

See users ids 110 (which I manually fixed), 122, 124, & 126.  118 and 119 are likely too.  I'm seeing skips in the user ID sequence which makes me think the toast error submission advances the sequence, but doesn't write the data.

Doesn't seem related to the Org length either as I've fixed one of them in question and have tried other orgs too.

**Comment by aaronstephenson** [comment]
OK, I'll take a look.

The skips in the IDs are from the just-created users being deleted when a down-stream error is encountered:
```
ucr_serializer = UserChangeRequestSerializer(data=user_change_request)
if ucr_serializer.is_valid():
    ucr_serializer.save()
else:
    NotificationCueStandard.objects.filter(created_by=user.id).delete()
    NotificationCuePreference.objects.filter(created_by=user.id).delete()
    user.delete()
    raise serializers.ValidationError(jsonify_errors(ucr_serializer.errors))
```

**Comment by aaronstephenson** [comment]
@nbaertlein what were the requested orgs and roles for the users you referenced above?

**Comment by nbaertlein** [comment]
User ID | Requested Role | Requested Organization | Requested Organization ID
-- | -- | -- | --
126 | PartnerAdmin | U.S. Geological Survey - National Wildlife Health Center | 2
124 | PartnerManager | Nevada Department of Wildlife - Las Vegas | 598
122 | PartnerManager | Southeastern Cooperative Wildlife Disease Study - University   of Georgia | 914
119 | unknown - Probably ParnterManager | unknown - prob Southeastern Cooperative Wildlife Disease Study   - University of Georgia | unknown - prob 914
118 | unknown | unknown | unknown
110 | PartnerManager | Southeastern Cooperative Wildlife Disease Study - University   of Georgia | 914

**Comment by aaronstephenson** [comment]
Through some more testing, I'm almost 100% certain that the problem lies with the comment. The client app says it is optional, and the API defines the serializer comment field as `required=False`, but DRF was _still_ requiring it, and returning an error saying so (the toast message you saw). Thus you were seeing the user record and notifications successfully created, but no user change request record or notifications created, because the comment is not evaluated until it gets to the UserChangeRequestSerializer, after the User is already created.

I did some digging, and it turns out this DRF `required=False` serializer field setting has tripped up a lot of people. Apparently it isn't enough to just set required to False, you also need to set `allow_blank=True`.  ¯\\\_(ツ)\_/¯

So, I did that (implemented in feb44927), and now I am getting the user change request notifications in my local dev and test environments.

@nbaertlein give it a try for yourself and let me know what you see.

**Comment by nbaertlein** [comment]
We're still having issues with this one.  We had 2 users sign up recently and they were getting errors and had to click again to get it to work.  The most recent was user 'elankau' who clicked submit 4 times before it successfully created account.  Still no user change request though.  Emily provided the following in the partner form:

Role - PartnerAdmin
Org - NWHC
Comment - yes - I included a note to you that this was set up at the highest level to be a test account so I could see everything. I figured you would down grade my access if that wasn't appropriate.

**Comment by nbaertlein** [comment]
Additional details from user 'nkudla':
I only remember trying to register once. 
1) Natural Resources Mgt Pro.
2) Partner role.
Org: NWHC
Comments: *none

**Comment by aaronstephenson** [comment]
It seems the cause of this bug was the definition of the `comments` field (distinct from `comment`) on the UserChangeRequestSerializer. The serializer was expecting the `comments` field to be submitted in the POST, which is not what we want at all (it's supposed to be a read-only field that lists existing comments). The fix was to update the field definition to include `read_only=True`.

Implemented in f0a098fe

**Comment by nbaertlein** [comment]
@aaronstephenson I just reran 4 new account request tests on the test site (see user id's 144, 145, 146, 147) and 1 new account did not register in the 'userchangerequest' table (and subsequent notifications). I think it may have been due to an initial token error due to my having the web API open. I think we can call this one closed, but I'll keep monitoring. Below were the scenarious run:

Test 1: Success
Username: neilaug1
Org: Alabama Department of Agriculture & Industries
Role: partner user
Comment: “new account test”

Test 2: Fail
Username: neilaug2
Org: Alabama Department of Agriculture & Industries - Alabama Veterinary Diagnostic Laboratories
Role: partner manager
Comment: (left blank)
other: had initial token fail due to being logged into web API

Test 3: Success
Username: neilaug3
Org: Alabama Department of Agriculture & Industries - Alabama Veterinary Diagnostic Laboratories
Role: partner manager
Comment: “new account test”

Test 4: Success
Username: neilaug4
Org: Alabama Department of Agriculture & Industries - Alabama Veterinary Diagnostic Laboratories
Role: partner manager
Comment: (left blank)
